kubetest2/pkg/build/build.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package build implements a common system for building kubernetes for deployers to use.
+package build
+
+import (
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	target = "bazel-release"
+)
+
+var (
+	// This will need changed to support other platforms.
+	tarballsToExtract = []string{
+		"kubernetes.tar.gz",
+		"kubernetes-test-linux-amd64.tar.gz",
+		"kubernetes-test-portable.tar.gz",
+		"kubernetes-client-linux-amd64.tar.gz",
+	}
+)
+
+// Build builds kubernetes with the bazel-release target
+func Build() error {
+	// TODO(RonWeber): This needs options
+	goPath := os.Getenv("GOPATH")
+	k8ssrc := filepath.Join(goPath, "src", "k8s.io", "kubernetes")
+	c := exec.Command("make", "-C", k8ssrc, target)
+	inheritOutput(c)
+	err := c.Run()
+	if err != nil {
+		return err
+	}
+	return extractBuiltTars()
+}
+
+// Stage stages the build to GCS.
+// Essentially release/push-build.sh --bucket=B --ci --gcs-suffix=S --noupdatelatest
+func Stage(location string) error {
+	re := regexp.MustCompile(`^gs://([\w-]+)/(devel|ci)(/.*)?`)
+	mat := re.FindStringSubmatch(location)
+	if mat == nil {
+		return fmt.Errorf("Invalid stage location: %v. Use gs://bucket/ci/optional-suffix", location)
+	}
+	bucket := mat[1]
+	ci := mat[2] == "ci"
+	gcsSuffix := mat[3]
+
+	args := []string{
+		"--nomock",
+		"--verbose",
+		"--noupdatelatest",
+		fmt.Sprintf("--bucket=%v", bucket),
+	}
+	if len(gcsSuffix) > 0 {
+		args = append(args, fmt.Sprintf("--gcs-suffix=%v", gcsSuffix))
+	}
+	if ci {
+		args = append(args, "--ci")
+	}
+
+	name, err := K8sDir("release", "push-build.sh")
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command(name, args...) //Not using kubetest2/pkg/exec because we need to set cmd.Dir.
+	cmd.Dir, err = K8sDir("kubernetes")
+	inheritOutput(cmd)
```

Comment:
probably cleaner: `cmd := inheritOutput(exec.Command(name, args...))`, and make inheritOutput return cmd

Reply:
Done.

Reply:
(looks like you did it below, but not for here?)